janitor/functions/transform_columns.py

```diff
@@ -121,95 +118,97 @@ def transform_columns(
+        2  1.176091  0.954243   3.0
+
+    Example: Using the `suffix` parameter to create new columns.
+
+        >>> df.transform_columns(["col1", "col3"], np.log10, suffix="_log")
+           col1  col2  col3  col1_log  col3_log
+        0     5     3    10  0.698970       1.0
+        1    10     6   100  1.000000       2.0
+        2    15     9  1000  1.176091       3.0
+
+    Example: Using the `new_column_names` parameter to create new columns.
+
+        >>> df.transform_columns(
+        ...     ["col1", "col3"],
+        ...     np.log10,
+        ...     new_column_names={"col1": "transform1"},
+        ... )
+           col1  col2  col3  transform1
+        0     5     3   1.0    0.698970
+        1    10     6   2.0    1.000000
+        2    15     9   3.0    1.176091
 
     :param df: A pandas DataFrame.
     :param column_names: An iterable of columns to transform.
     :param function: A function to apply on each column.
-    :param suffix: (optional) Suffix to use when creating new columns to hold
+    :param suffix: Suffix to use when creating new columns to hold
         the transformed values.
     :param elementwise: Passed on to `transform_column`; whether or not
         to apply the transformation function elementwise (True)
         or columnwise (False).
-    :param new_column_names: (optional) An explicit mapping of old column names
-        to new column names.
+    :param new_column_names: An explicit mapping of old column names in
+        `column_names` to new column names. If any column specified in
+        `column_names` is not a key in this dictionary, the transformation
+        will happen in-place for that column.
     :returns: A pandas DataFrame with transformed columns.
-    :raises ValueError: if both `suffix` and `new_column_names` are
-        specified
-    """
-    dest_column_names = dict(zip(column_names, column_names))
-
+    :raises ValueError: If both `suffix` and `new_column_names` are
+        specified.
+    """  # noqa: E501
     check("column_names", column_names, [list, tuple])
 
     if suffix is not None and new_column_names is not None:
         raise ValueError(
-            "only one of suffix or new_column_names should be specified"
+            "Only one of `suffix` or `new_column_names` should be specified."
         )
 
-    if suffix:  # If suffix is specified...
+    if suffix:
         check("suffix", suffix, [str])
-        for col in column_names:
-            dest_column_names[col] = col + suffix
-
-    if new_column_names:  # If new_column_names is specified...
+        dest_column_names = {col: col + suffix for col in column_names}
+    elif new_column_names:
         check("new_column_names", new_column_names, [dict])
         dest_column_names = new_column_names
+    else:
+        dest_column_names = {}
 
-    # Now, transform columns.
-    for old_col, new_col in dest_column_names.items():
+    for old_col in column_names:
         df = transform_column(
```

Comment:
multiple copies here, since it is calling transform_column (each assign will create a copy). Is it possible to do it once? again just a thought, no concrete alternatives

Reply:
True, I did a bit of refactoring. Kindly re-review the `transform_columns.py` file again, thanks!